Update README.md

aob93 opened the pull request.

aob93 committed:
Update README.md

aob93 committed:
Update README.md

leej3 commented:
have a think about how to handle two people on the same project. There are details to using git in conjunction with another that can be tricky. It's a useful skill and well worth learning but will slow you down lots. Duplicating efforts across two repositories is not especially desirable either though. If you do choose to both use the same repo I can walk you through how to handle merge conflicts etc.

leej3 commented:
@hsowards

aob93 committed:
Update README.md

aob93 committed:
Update README.md

leej3 commented:
Do you have code written? If so, please push it to GitHub so that we can help out.

aob93 commented:
Hi John,

Thanks for getting in touch. Code is all written, just getting it uploaded now.

My code takes a list of paths to files stored in our shared drives. Should I also upload these files to github? (some are quite large)

~Aidan

From: john lee <notifications@github.com>
Reply-To: biof309/project_spring_2020 <reply@reply.github.com>
Date: Wednesday, May 6, 2020 at 12:12 PM
To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
Cc: aob93 <aidanobrien93@hotmail.com>, Author <author@noreply.github.com>
Subject: Re: [biof309/project_spring_2020] Update README.md (#7)


Do you have code written? If so, please push it to GitHub so that we can help out.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-624742924&data=02%7C01%7C%7Cf6c5b1fe79f846b396b708d7f1d833c7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243783195582213&sdata=68tzgZlzpXvzmOI7J1l1uIv1EAteN7Non%2FYy3Abai3c%3D&reserved=0>, or unsubscribe<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOJ3M34TCPB6GJFN7HDRQGD43ANCNFSM4K5GOKZA&data=02%7C01%7C%7Cf6c5b1fe79f846b396b708d7f1d833c7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243783195592207&sdata=dCUqpkMNn8Zu2GDXbbwcFFO6itvH14WsV9hX137jGqU%3D&reserved=0>.

leej3 commented:
Great.

And no don't upload large files. A few small sample files/directory tree in
the tests directory so that you can write tests but otherwise no, git repos
shouldn't have large data files in them.

On Wed, May 6, 2020 at 12:15 PM aob93 <notifications@github.com> wrote:

> Hi John,
>
> Thanks for getting in touch. Code is all written, just getting it uploaded
> now.
>
> My code takes a list of paths to files stored in our shared drives. Should
> I also upload these files to github? (some are quite large)
>
> ~Aidan
>
> From: john lee <notifications@github.com>
> Reply-To: biof309/project_spring_2020 <reply@reply.github.com>
> Date: Wednesday, May 6, 2020 at 12:12 PM
> To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
> Cc: aob93 <aidanobrien93@hotmail.com>, Author <author@noreply.github.com>
> Subject: Re: [biof309/project_spring_2020] Update README.md (#7)
>
>
> Do you have code written? If so, please push it to GitHub so that we can
> help out.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub<
> https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-624742924&data=02%7C01%7C%7Cf6c5b1fe79f846b396b708d7f1d833c7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243783195582213&sdata=68tzgZlzpXvzmOI7J1l1uIv1EAteN7Non%2FYy3Abai3c%3D&reserved=0>,
> or unsubscribe<
> https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOJ3M34TCPB6GJFN7HDRQGD43ANCNFSM4K5GOKZA&data=02%7C01%7C%7Cf6c5b1fe79f846b396b708d7f1d833c7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243783195592207&sdata=dCUqpkMNn8Zu2GDXbbwcFFO6itvH14WsV9hX137jGqU%3D&reserved=0>.
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/biof309/project_spring_2020/pull/7#issuecomment-624744610>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABJKZKBHQEHJIVFAIZFLBBDRQGEITANCNFSM4K5GOKZA>
> .
>

aob93 committed:
Update README.md

aob93 committed:
Add files via upload

aob93 commented:
Hi John,

All should be on the repository now.

All the best,
Aidan

From: john lee <notifications@github.com>
Reply-To: biof309/project_spring_2020 <reply@reply.github.com>
Date: Wednesday, May 6, 2020 at 12:19 PM
To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
Cc: aob93 <aidanobrien93@hotmail.com>, Author <author@noreply.github.com>
Subject: Re: [biof309/project_spring_2020] Update README.md (#7)

Great.

And no don't upload large files. A few small sample files/directory tree in
the tests directory so that you can write tests but otherwise no, git repos
shouldn't have large data files in them.

On Wed, May 6, 2020 at 12:15 PM aob93 <notifications@github.com> wrote:

> Hi John,
>
> Thanks for getting in touch. Code is all written, just getting it uploaded
> now.
>
> My code takes a list of paths to files stored in our shared drives. Should
> I also upload these files to github? (some are quite large)
>
> ~Aidan
>
> From: john lee <notifications@github.com>
> Reply-To: biof309/project_spring_2020 <reply@reply.github.com>
> Date: Wednesday, May 6, 2020 at 12:12 PM
> To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
> Cc: aob93 <aidanobrien93@hotmail.com>, Author <author@noreply.github.com>
> Subject: Re: [biof309/project_spring_2020] Update README.md (#7)
>
>
> Do you have code written? If so, please push it to GitHub so that we can
> help out.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub<
> https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-624742924&data=02%7C01%7C%7Cf6c5b1fe79f846b396b708d7f1d833c7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243783195582213&sdata=68tzgZlzpXvzmOI7J1l1uIv1EAteN7Non%2FYy3Abai3c%3D&reserved=0>,
> or unsubscribe<
> https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOJ3M34TCPB6GJFN7HDRQGD43ANCNFSM4K5GOKZA&data=02%7C01%7C%7Cf6c5b1fe79f846b396b708d7f1d833c7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243783195592207&sdata=dCUqpkMNn8Zu2GDXbbwcFFO6itvH14WsV9hX137jGqU%3D&reserved=0>.
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/biof309/project_spring_2020/pull/7#issuecomment-624744610>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABJKZKBHQEHJIVFAIZFLBBDRQGEITANCNFSM4K5GOKZA>
> .
>


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-624747228&data=02%7C01%7C%7C6a572d5cfdbb4594969608d7f1d94728%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243787816286734&sdata=jna5RwINgR7pMItB1AF1NzS3BsoXjKOfkYjJWbyRwsg%3D&reserved=0>, or unsubscribe<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOMCJ6ZAG5O6NK3HRU3RQGEZXANCNFSM4K5GOKZA&data=02%7C01%7C%7C6a572d5cfdbb4594969608d7f1d94728%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637243787816296728&sdata=Io%2BYsU9Xv9jDOZUNa%2B9mX8EfQbUZos2z8xH9PHFmVS4%3D&reserved=0>.

emilyyaklich committed:
commiting changes to new branch

emilyyaklich commented:
Hi Aidan,

I just created a pull request of possible updates to your project repository. I have created a setup.py file (in order to package your code) which you should look at and fill out the necessary information. Additionally I changed your "code" directory to "project_code" and your "annotation.plot.py" file to "annotation_plot.py". 

I have also added an example test file (tests.py) to your tests directory to help start you start writing tests for you code. The example test just confirms that the "header" that you read in is a list. If you type in 
```
pytest tests.py
```
 in your terminal you should be able to see that the test passes. 

These are just suggestions to help you start testing your code as well as package it.

Let me know if you have any questions!

aob93 commented:
Hi Emily,

Thanks so much for your help here. Much appreciated. I'll take a look asap.

Aidan

Sent from Outlook Mobile<https://aka.ms/blhgte>
________________________________
From: emily yaklich <notifications@github.com>
Sent: Thursday, May 7, 2020 12:33:16 AM
To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
Cc: aob93 <aidanobrien93@hotmail.com>; Author <author@noreply.github.com>
Subject: Re: [biof309/project_spring_2020] Update README.md (#7)


Hi Aidan,

I just created a pull request of possible updates to your project repository. I have created a setup.py file (in order to package your code) which you should look at and fill out the necessary information. Additionally I changed your "code" directory to "project_code" and your "annotation.plot.py" file to "annotation_plot.py".

I have also added an example test file (tests.py) to your tests directory to help start you start writing tests for you code. The example test just confirms that the "header" that you read in is a list. If you type in

pytest tests.py


in your terminal you should be able to see that the test passes.

These are just suggestions to help you start testing your code as well as package it.

Let me know if you have any questions!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-625023275&data=02%7C01%7C%7Cd4ceaee137174a8f3bec08d7f23fc358%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637244227975070329&sdata=jzYBaLdA%2F56ojvdelmsvgpV5N4U4Fi8YHgwtoJOlx3Q%3D&reserved=0>, or unsubscribe<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOJPU5FNTCC63KB2CDDRQI2YZANCNFSM4K5GOKZA&data=02%7C01%7C%7Cd4ceaee137174a8f3bec08d7f23fc358%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637244227975080329&sdata=l8AWcREBKzea1iowN4BPvYgqMX0gNb2DENt%2F4TnxrPA%3D&reserved=0>.

emilyyaklich committed:
delete unnecessary file

emilyyaklich committed:
update setup file with correct path

aob93 committed:
Merge pull request #1 from emilyyaklich/updates

Possible updates

emilyyaklich commented:
HI Aidan,

One more suggestion, I think it may be easier to write unit tests for you code if divide your code up into multiple functions, instead of the one main function.

Also, now that I think about it, it may be best to create a ''test_data" folder within your test directory (I basically just copied your data files over but it would be clearer to see a test directory with all of the data files) in order to organize your code. You can then read in the data from the directory as is currently done in the test. Also, I think a better test statement for the example test that I wrote is:
``` 
assert header[0] == 'Chr'
```
because this would actually test that you are reading in what you think you are reading in from the datafile.

aob93 commented:
OK this sounds sensible. Basically just do the same things for each object being used in the function?

From: emily yaklich <notifications@github.com>
Reply-To: biof309/project_spring_2020 <reply@reply.github.com>
Date: Thursday, May 7, 2020 at 6:15 PM
To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
Cc: aob93 <aidanobrien93@hotmail.com>, Author <author@noreply.github.com>
Subject: Re: [biof309/project_spring_2020] Update README.md (#7)


HI Aidan,

One more suggestion, I think it may be easier to write unit tests for you code if divide your code up into multiple functions, instead of the one main function.

Also, now that I think about it, it may be best to create a ''test_data" folder within your test directory (I basically just copied your data files over but it would be clearer to see a test directory with all of the data files) in order to organize your code. You can then read in the data from the directory as is currently done in the test. Also, I think a better test statement for the example test that I wrote is:

assert header[0] == 'Chr'

because this would actually test that you are reading in what you think you are reading in from the datafile.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://apc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-625524193&data=02%7C01%7C%7C398c04706b824e1df2ec08d7f2d41a94%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637244865098017421&sdata=FLJ36By1Y%2FnaXbCCAUWdiqesI8kF7t%2BT%2BE%2BvgIjaciU%3D&reserved=0>, or unsubscribe<https://apc01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOL3EWENMEC2CROGNILRQMXGTANCNFSM4K5GOKZA&data=02%7C01%7C%7C398c04706b824e1df2ec08d7f2d41a94%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637244865098027406&sdata=9AjVkWKZ8FdDeUQTTwjcPw2V5kuGZtRav4AIdSkJJqg%3D&reserved=0>.

emilyyaklich commented:
Yes, basically, you want to test that your code is doing what you think it is doing with the tests.

aob93 committed:
Update tests.py

aob93 committed:
Update tests.py

aob93 committed:
Update tests.py

emilyyaklich commented:
I have a few more comments that may be useful: 

I think you also need to remove the 'sample_tests.py' from your test directory as there is a failing test in this script so it is causing issues. 

Additionally, filling out your package details in the 'setup.py' script (name, email, package name, etc.) will also be useful for you.

aob93 committed:
Delete sample_test.py

aob93 committed:
Update setup.py

aob93 commented:
Thank you Emily, I’ll get onto it.

Aidan

From: emily yaklich <notifications@github.com>
Reply-To: biof309/project_spring_2020 <reply@reply.github.com>
Date: Thursday, May 7, 2020 at 8:02 PM
To: biof309/project_spring_2020 <project_spring_2020@noreply.github.com>
Cc: aob93 <aidanobrien93@hotmail.com>, Author <author@noreply.github.com>
Subject: Re: [biof309/project_spring_2020] Update README.md (#7)


I have a few more comments that may be useful:

I think you also need to remove the 'sample_tests.py' from your test directory as there is a failing test in this script so it is causing issues.

Additionally, filling out your package details in the 'setup.py' script (name, email, package name, etc.) will also be useful for you.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbiof309%2Fproject_spring_2020%2Fpull%2F7%23issuecomment-625556869&data=02%7C01%7C%7Ccf510b4554774d6fa4d508d7f2e3291f%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637244929762557749&sdata=LEve4vlMRGmzvo5vgVNknKl%2F0C15Y8%2FBVTLJKjv9EKQ%3D&reserved=0>, or unsubscribe<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAOPSSOLEXZVPRZBGWDP3S6TRQND25ANCNFSM4K5GOKZA&data=02%7C01%7C%7Ccf510b4554774d6fa4d508d7f2e3291f%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637244929762567742&sdata=t3lruHtJBGTDYS3hsXi6AVd%2BDiczuXcTxUr%2BWklniLY%3D&reserved=0>.

emilyyaklich commented:
Hi Aidan,

Well done on the project. You show a good understanding of Python and the use of NumPy and Pandas is nice as well as the creation of a plot using matplotlib. In the future, I would suggest modularizing your code into multiple functions which will then make it easier to debug as you think about writing tests. Also, I suggest adding files like *.ipynb_checkpoints* to your *.gitignore* file. Overall, the project is great, good work! 

Cheers,

Emily
